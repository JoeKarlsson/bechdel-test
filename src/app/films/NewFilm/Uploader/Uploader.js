import React from 'react';
import ErrorBoundary from '../../../shared/ErrorBoundary/ErrorBoundary';

import './Uploader.scss';

const Uppy = require('uppy/lib/core');
const XHRUpload = require('uppy/lib/plugins/XHRUpload');
const DragDrop = require('uppy/lib/react/DragDrop');

const uppy = Uppy({
	meta: { type: 'avatar' },
	restrictions: { maxNumberOfFiles: 1 },
	autoProceed: true,
});

uppy.use(XHRUpload, {
	endpoint: '/api/film',
	method: 'post',
	formData: true,
	limit: 1,
});

uppy.on('complete', result => {
	const url = result.successful[0].uploadURL;
	console.log('url', url);
});

uppy.run();

const Uploader = () => {
	return (
		<div>
			<ErrorBoundary>
				<DragDrop
					uppy={uppy}
					locale={{
						strings: {
							chooseFile: 'Pick a new avatar',
						},
					}}
				/>
			</ErrorBoundary>
		</div>
	);
};

export default Uploader;
