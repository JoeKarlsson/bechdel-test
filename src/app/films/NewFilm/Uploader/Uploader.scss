@import 'uppy/dist/uppy.min.css';
