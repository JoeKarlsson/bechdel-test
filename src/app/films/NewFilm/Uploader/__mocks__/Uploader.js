import React from 'react';

const mockComponent = () => <div>Mock Uploader</div>;

export default mockComponent;
