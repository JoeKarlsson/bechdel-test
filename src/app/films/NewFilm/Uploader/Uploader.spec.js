/*
	eslint-disable class-methods-use-this
*/

import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import Uploader from './Uploader';

jest.mock('uppy/lib/core', () => {
	class UppyClass {
		use() {
			return jest.fn();
		}
		on() {
			return jest.fn();
		}
		run() {
			return jest.fn();
		}
	}

	const Uppy = () => {
		return new UppyClass();
	};

	return Uppy;
});
jest.mock('uppy/lib/plugins/XHRUpload');
jest.mock('uppy/lib/react/DragDrop', () => {
	const DragDrop = () => <div>Drag and Drop</div>;
	return DragDrop;
});

configure({ adapter: new Adapter() });

describe('Uploader', () => {
	let wrapper;

	global.requestAnimationFrame = callback => {
		setTimeout(callback, 0);
	};

	beforeEach(() => {
		wrapper = shallow(<Uploader />);
	});

	describe('rendering', () => {
		describe('initial state', () => {
			it('is rendered', () => {
				const component = renderer.create(<Uploader />);
				const tree = component.toJSON();
				expect(tree).toMatchSnapshot();
			});

			it('is rendered correctly', () => {
				expect(wrapper).toHaveLength(1);
			});

			it('should have correct inital instance', () => {
				const initialInstance = wrapper.instance();
				const expectedInstance = null;
				expect(initialInstance).toBe(expectedInstance);
			});
		});
	});
});
