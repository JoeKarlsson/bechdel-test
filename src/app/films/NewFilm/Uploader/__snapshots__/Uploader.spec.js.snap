// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Uploader rendering initial state is rendered 1`] = `
<div>
  <div>
    Drag and Drop
  </div>
</div>
`;
